Assets/BossRoom/Scripts/Server/Game/Action/ActionPlayer.cs

```diff
@@ -109,24 +112,37 @@ private void StartAction()
         {
             if (m_Queue.Count > 0)
             {
+                float reuseTime = m_Queue[0].Description.ReuseTimeSeconds;
+                if (   reuseTime > 0
```

Comment:
I'm not 100% sure if this is the correct place to check but I did encounter a bug with the mage bolt.

If you spam left & right click simultaneously you'll see that the animation & sound fire. but no bolt is created.

Reply:
Nice find! This is a bug in the input-processing code path, and not this PR per se. It happens with all classes' basic attacks, most easily noticed with the archer, who does the animation but no projectile comes out. I think this is also happening to the tank and rogue's melee attacks, and contributes to why those attacks don't always seem to "hit": they are actually being cancelled almost instantly after they begin, but the animation trigger has already fired. 

(Prior to this PR, the client didn't correctly cancel the mage's graphics, so the bolt was rendered on the client even though it had already been canceled on the server.)

Can you open a bug on this? (I don't have direct bug access.) I'd like to fix this in a separate PR if possible. I hacked in a quick-fix locally, but I'm not confident that it doesn't cause other problems. Will need to get David's feedback on the proper solution.

Reply:
oh right, @dwoodruffsf ^^ FYI on a bug that needs fixing

Reply:
I made a bug [task](https://unity3d.atlassian.net/browse/GOMPS-467) to track this.